### DISCUSSION TITLE
Moving commit diverges branches

### DISCUSSION THREAD

**Discussion by HamidKadja** [community/community#23999] [New to GitHub]
<p>I have a repo with a master branch of 4 commits:</p>
<p><strong>1-2-3-4 (HEAD → master)</strong></p>
<p>Then I created a new branch (b2) and made the fifth commit, I merged b2 to master. <code>$ git merge b2</code> and I got this log :</p>
<p><strong>1-2-3-4-5(HEAD → master, b2)</strong></p>
<p>Now I wanna change the order of commit 2 so I made <code>$ git rebase -i HEAD~5</code>, and I got this log :</p>
<p><strong>1-3-4-2-5 (HEAD → master) <br>  \</strong><br>
<strong>2-3-4-5 (b2)</strong></p>
<p>Two diverge branches.</p>
<p>And the result I want is :</p>
<p><strong>1-3-4-2-5 (HEAD → master, b2)</strong></p>

**Comment by jsoref** [comment, marked as answer]
<p>There are various ways to achieve this:</p>
<ul>
<li><code>git push origin master:b2 --force-with-lease</code></li>
<li><code>git checkout b2; git reset --hard master; git push origin b2 --force-with-lease</code></li>
</ul>

**Comment by HamidKadja** [comment]
<p>Thank you, this worked.</p>
